src/node_contextify.cc

```diff
@@ -621,7 +621,7 @@ class ContextifyScript : public BaseObject {
         new ContextifyScript(env, args.This());
 
     TryCatch try_catch(env->isolate());
-    Local<String> code = args[0]->ToString(env->isolate());
+    Local<String> code = args[0]->ToString(env->context()).ToLocalChecked();
```

Comment:
This change is actually incorrect, I think (didn't try this out yet, so I might be wrong):

```js
new vm.Script({
  toString() {
    throw new Error();
  }
});
```

would make `ToString()` return an empty `MaybeLocal`, right? That's fine because we have a `try_catch` here to check for those kinds of conditions.

So I think the two options are:

- Use `.FromMaybe(Local<String>())` to get the previous behaviour
- Move this line to before the `TryCatch` and return from the function immediately if the result is empty

Either way, it would be nice if you could add a test that the code snippet above doesn't crash the process :)

Reply:
> would make ToString() return an empty MaybeLocal, right?

Yes it is.
I confirmed that it will crash when running that code.

```
./node test.js
FATAL ERROR: v8::ToLocalChecked Empty MaybeLocal.
 1: node::Abort() [/Users/leko/.ghq/github.com/Leko/node/./node]
 2: node::FatalException(v8::Isolate*, v8::Local<v8::Value>, v8::Local<v8::Message>) [/Users/leko/.ghq/github.com/Leko/node/./node]
 3: v8::V8::ToLocalEmpty() [/Users/leko/.ghq/github.com/Leko/node/./node]
 4: node::(anonymous namespace)::ContextifyScript::New(v8::FunctionCallbackInfo<v8::Value> const&) [/Users/leko/.ghq/github.com/Leko/node/./node]
 5: v8::internal::FunctionCallbackArguments::Call(void (*)(v8::FunctionCallbackInfo<v8::Value> const&)) [/Users/leko/.ghq/github.com/Leko/node/./node]
 6: v8::internal::MaybeHandle<v8::internal::Object> v8::internal::(anonymous namespace)::HandleApiCallHelper<true>(v8::internal::Isolate*, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::HeapObject>, v8::internal::Handle<v8::internal::FunctionTemplateInfo>, v8::internal::Handle<v8::internal::Object>, v8::internal::BuiltinArguments) [/Users/leko/.ghq/github.com/Leko/node/./node]
 7: v8::internal::Builtin_Impl_HandleApiCall(v8::internal::BuiltinArguments, v8::internal::Isolate*) [/Users/leko/.ghq/github.com/Leko/node/./node]
 8: 0x32b64f842fd
Abort trap: 6
```

So I added test case and fix it.
Please check it again.